File: backend/src/main/java/com/example/backend/domain/model/Book.java
package com.example.backend.domain.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity(name = "books")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    private String title;
    private String author;
    private String genre;
    private String description;
    private String publicationYear;
    private String status;
    private String readingMonth;
    private String readingYear;
    private String score;

    @ManyToOne
    private User user;

}

File: frontend/src/types/Book.ts
export type Book = {
  id?: number;
  title: string;
  author: string;
  genre: string;
  publicationYear: string;
  description: string;
  status: "WANT" | "READING" | "READ";
  readingYear: string;
  readingMonth: string;
  score: string;
};